src/VerticalGridDelegate.js

```diff
@@ -262,23 +262,19 @@ kind.extendMethods(p, {
 	* @private
 	*/
 	adjustBuffer: function (list) {
-		var pc = this.pageCount(list),
-			ds = this.defaultPageSize(list),
+		var cs = this.childSize(list),
+			count = list.collection ? list.collection.length : 0,
 			bs = 0, sp = list.psizeProp, ss = list.ssizeProp,
-			n = list.$.buffer.node || list.$.buffer.hasNode(), p;
+			n = list.$.buffer.node || list.$.buffer.hasNode();
 		if (n) {
-			for (var i=0; i<pc; ++i) {
-				p = list.metrics.pages[i];
-				bs += (p && p[sp]) || ds;
-			}
-			bs += list.spacing;
+			bs = (Math.floor(count/list.columns)+(count%list.columns? 1: 0))*cs;
```

Comment:
To be honest, I don't really understand what (count%list.columns? 1: 0) doing in this calculation. I borrowed this line from pageHeight function to calculate buffer size. So, if you say that then I think we could think simplify pageHeight also. :)